test/functional/legacy/autocmd_option_spec.lua

```diff
@@ -0,0 +1,120 @@
+local helpers = require('test.functional.helpers')
+local clear, nvim, eq = helpers.clear, helpers.nvim, helpers.eq
+local source, execute = helpers.source, helpers.execute
+
+describe('au OptionSet', function()
+  describe('with * as <amatch>', function()
+    describe('matches when being set any option', function()
```

Comment:
You should do something with test names: this way they look like “au OptionSet with * as <amatch> ***matches*** when being set any option ***should set*** several global list and number option” which is incorrect. Note the naming: first argument to `describe` should show what tests describes and it should not contain any verbs. This is basically why function is named `describe`. Functions `it` actually describe the behaviour, `it` plus first argument should read like a sentence. So “matches” can only be in a `it` argument, but not here or you will get weird results. I do not really see why you need this first `describe` call, two `describe`s are enough.